### DISCUSSION TITLE
document.getElementsByClassName("classname") not working

### DISCUSSION THREAD

**Discussion by nrschoen** [jspsych/jsPsych#1166] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/65179042/97997010-cafb0a00-1de8-11eb-817b-55c5c770f026.png)

I am getting the following error when trying to change properties of a sample box. Similar code works fine in JSFiddle so I am wondering what I am doing wrong.

![image](https://user-images.githubusercontent.com/65179042/97997155-f67df480-1de8-11eb-8b2b-05a169c7c1e4.png)

Thank you in advance for your help!
Neil

**Comment by jodeleeuw** [comment, marked as answer]
Hi Neil,

(Thanks for opening this as a new thread!)

The issue is that the box doesn't exist on the page until after the experiment has begun. When you call `getElementsByClassName` on line 4, there is no box on the page because the experiment hasn't started running yet.

You can move the code into the `on_load` event handler of the plugin. This is an event that is triggered once the plugin has added its content to the page.

```js
var box = {
	type: 'html-keyboard-response',
	stimulus: '<div class="box"></div>',
	on_load: function(){
		var box_el = document.getElementsByClassName('box')[0];
		box_el.style.backgroundColor = "green";
	}
}
```

**Reply by nrschoen** [reply]
Great! Thanks for the quick reply!
